### ISSUE TITLE
Font support for unicode characters?

### ISSUE THREAD

**Issue by riimi** [EngoEngine/engo#678]
Hi! I am new to engo, into this these days.
I have been trying to draw a string containing unicode characters: Korean, Chinese, Japanese.
Do I have to implement a new shader for this?
Thanks in advance!

**Comment by Noofbiz** [comment]
The font shader should be able to handle it, but right now it's limited to the first 200 glyphs from the font [here](https://github.com/EngoEngine/engo/blob/6f88a7379a25d81e976cf6e86ea3617f3b480b1a/common/font.go#L301). This is so that generating a font atlas happens quickly and gets everything you need, but also because I've never taken the time to use fonts containing those characters. I think the easiest fix would be to allow users to set the number of Glyphs used, and have it default to 200 if they don't.

**Comment by riimi** [comment]
I solved the problem by increasing [UnicodeCap](https://github.com/EngoEngine/engo/blob/6f88a7379a25d81e976cf6e86ea3617f3b480b1a/common/render_shaders.go#L17) according to your advice. But, the solution seems inefficient way because the font makes the big font atlas starting from index 0. I think I will have to find more efficient way to support multiple languages.

**Comment by Noofbiz** [comment]
Yeah I'm thinking of adding to fonts a []int that will let you decide which code points are used. If empty it'll do 0-200 for backwards compatibility. I think this is best because then if you only need 8 characters it'll only make 8 characters and not waste memory on the other ones.

**Comment by kayon** [comment]
```
var glyphBoundsCache = map[font.Face]map[rune]fixed.Rectangle26_6{}

func getGlyphBounds(face font.Face, r rune) fixed.Rectangle26_6 {
	if _, ok := glyphBoundsCache[face]; !ok {
		glyphBoundsCache[face] = map[rune]fixed.Rectangle26_6{}
	}
	if b, ok := glyphBoundsCache[face][r]; ok {
		return b
	}
	b, _, _ := face.GlyphBounds(r)
	glyphBoundsCache[face][r] = b
	return b
}
```
Maybe you can do it like ebiten.
